test/junit/scala/issues/OptimizedBytecodeTest.scala

```diff
@@ -56,9 +55,9 @@ class OptimizedBytecodeTest extends ClearAfterClass {
     val List(c) = compileClasses(compiler)(code)
 
     assertSameSummary(getSingleMethod(c, "t"), List(
-      LDC, ASTORE, ALOAD /*0*/, ALOAD /*1*/, "C$$$anonfun$1", IRETURN))
-    assertSameSummary(getSingleMethod(c, "C$$$anonfun$1"), List(LDC, "C$$$anonfun$2", IRETURN))
-    assertSameSummary(getSingleMethod(c, "C$$$anonfun$2"), List(-1 /*A*/, GOTO /*A*/))
+      LDC, ASTORE, ALOAD /*0*/, ALOAD /*1*/, "$anonfun$t$1", IRETURN))
+    assertSameSummary(getSingleMethod(c, "$anonfun$t$1"), List(ALOAD, IFNONNULL /*8*/, ACONST_NULL, ATHROW, -1 /*8*/, LDC, "$anonfun$t$2", IRETURN))
```

Comment:
We could avoid the null check by teaching the backend that the `$this` parameter is non-nullable.